diff -ruN src/plugins/Transports/http/httpinputfactory.cpp src-new/plugins/Transports/http/httpinputfactory.cpp
--- src/plugins/Transports/http/httpinputfactory.cpp	2016-01-27 06:18:19.000000000 +0000
+++ src-new/plugins/Transports/http/httpinputfactory.cpp	2016-03-13 00:13:25.790082278 +0000
@@ -30,7 +30,7 @@
 const InputSourceProperties HTTPInputFactory::properties() const
 {
     InputSourceProperties p;
-    p.protocols << "http";
+    p.protocols << "http" << "https";
     p.name = tr("HTTP Plugin");
     p.shortName = "http";
     p.hasAbout = true;
diff -ruN src/plugins/Transports/http/httpstreamreader.cpp src-new/plugins/Transports/http/httpstreamreader.cpp
--- src/plugins/Transports/http/httpstreamreader.cpp	2016-01-27 06:18:19.000000000 +0000
+++ src-new/plugins/Transports/http/httpstreamreader.cpp	2016-03-13 00:13:39.890082578 +0000
@@ -351,7 +351,7 @@
         setErrorString(errorBuffer);
         emit error();
     }
-    QIODevice::close();
+    // QIODevice::close();
 }
 
 qint64 HttpStreamReader::readBuffer(char* data, qint64 maxlen)
